src/mesh/domain.hpp

```diff
@@ -140,6 +142,23 @@ class IndexShape {
     }
   }
 
+  KOKKOS_INLINE_FUNCTION
+  static IndexShape GetOnDevice(int nx3, int nx2, int nx1, int ng) {
+    IndexShape out;
+    int idx = 0;
+    for (auto nx : {nx1, nx2, nx3}) {
+      if (nx == 0) {
+        out.x_[idx] = IndexRange{0, 0};
+        out.entire_ncells_[idx] = 1;
+      } else {
+        out.x_[idx] = IndexRange{ng, (ng + nx - 1)};
+        out.entire_ncells_[idx] = nx + 2 * ng;
+      }
+      idx++;
+    }
+    return out;
+  }
```

Comment:
Honestly, I don't completely remember why I needed this. I think I probably wanted to construct and `IndexShape` on device directly and the constructor included something that wouldn't let me mark it as `KOKKOS_INLINE_FUNCTION`.

Reply:
Maybe we should just make the constructor work like this one then?

Reply:
The constructor takes a vector containing the sizes of the dimensions and then uses that in a number of places (and does things like reduce its order). There was enough going on that I was hesitant to dive into refactoring that code so that it would work on GPU and making sure I got all of the edge cases in there. I don't think there is currently any guarantee that `GetOnDevice` returns exactly the same thing as calling the constructor with the same dimension sizes.

Reply:
Yeah okay... Maybe a different name than `GetOnDevice` then? Like `MakeWithoutVectors` or something?

Reply:
Change to `GetFromSeparateInts`. Not a huge fan of that name, but couldn't come up with something better.